packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -166,15 +176,21 @@ export const Table = memo((props: Props) => {
     }
 
     const footerItems = getFooterItems(
-      headerGroups[0].headers as unknown as Array<{ field: Field }>,
+      /*
+        The `headerGroups` object is NOT based on the `data.fields`, but instead on the currently rendered headers in the Table,
+        which may or may not include the Row Numbers column.
+      */
+      headerGroups[0].headers as unknown as Array<{ id: string; field: Field }>,
+      // The `rows` object, on the other hand, is based on the `data.fields` data, and therefore ALWAYS include the Row Numbers column data.
       createFooterCalculationValues(rows),
       footerOptions,
       theme
     );
 
     if (isCountRowsSet) {
       const footerItemsCountRows: FooterItem[] = new Array(footerItems.length).fill(undefined);
-      footerItemsCountRows[0] = data.length.toString();
```

Comment:
Think this should also be reverted.

Looking through it though, I realised there are a lot of improvements I could do here for the count rows stuff. Like here, for example, this entire if-clause could be moved before the `getFooterItems` call and return if count rows is set. We don't necessarily need `footerItems.length`